model/Task/TaskCategoriserInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoTaskQueue\model\Task;
+
+/**
+ * An interface for any task to specify its own category.
+ *
+ * @author Gyula Szucs <gyula@taotesting.com>
+ */
+interface TaskCategoriserInterface
```

Comment:
We will need migration script? Because when you assign a category for a Task, the previous exists task logs we will not have that category.
I think we should create a config for this mapping and just read the category from there, because we are going to define category for a task then we realize that it's not the right one (business decision) and we will have to update the entire table to match with the new change requested.
TaskLogEntity can be seeded from different sources(database, config) etc.